### DISCUSSION TITLE
signalcaptcha

### DISCUSSION THREAD

**Discussion by RS-eco** [AsamK/signal-cli#500] [Q&A] [answered]
Great, I've added a wiki page for it. Feel free to update it if something's missing or unclear:
https://github.com/AsamK/signal-cli/wiki/Registration-with-captcha

_Originally posted by @AsamK in https://github.com/AsamK/signal-cli/issues/495#issuecomment-773542134_

**Comment by RS-eco** [comment]
Thanks for the wiki, however I still struggle to find the "signalcaptcha://CODE" in the browser development tools. :/
All I can find using Firefox under Network is: signalcaptcha://" + token in the generate.html file. Could you maybe clarify in which file the code is located and how the CODE should look like, i.e. number of digits/letters, ...

Thanks a lot.

**Comment by AsamK** [comment, marked as answer]
~The signalcaptcha:// is sent in the `Location` header of the response.~
After filling out the captcha, in the developer tools console tab you should see a failed redirect to signalcaptcha://

**Reply by RS-eco** [reply]
Sorry, I might just be too stupid for this. But I still struggle to find it. I am looking under the Network Tab, search for signalcaptcha:// and find three files. One of them is the generate.html file. There I can look under HEADER or under RESPONSE, but nowhere do I find the token and I also struggle to find Location. All I find is this:
`window.location = "signalcaptcha://" + token;`
but there no token is displayed.

**Reply by AsamK** [reply]
ok, I rechecked, it's not in the network tab, because it's redirected by javascript and not HTTP header.
After filling out the captcha, in the developer tools console tab you should see a failed redirect, like this:
`Navigation vers « signalcaptcha://03AGdBq249J34MPgcoSA_Omo9OA6hj5mHUu65Aucb-rMWJotcZ9fYveVoKwsX9f3sEp98rYfbyrW8KK9D3By13Sq7VAe84bvBTd8sTmWZiDsi3cRaOVadEoMSjPj8Pz3t6GrLM7mE6rFSJNXl0YdWBa-UNcxCGsu8von3asNil8lj10cYXbJHCuTuYhGUU15dkRRBCj9xi_aRnXHTbM9gq8Frioc5Y0x_vysLl7w1RWm0DqZUjhwb1lLutNoZUzbUQ0MyetIFgr-MxffylWxCdJ7XwJtUrIjvr_Oqhw0aw023b29ssDb_uQEP3gdeG0mJ334wxu2XH5KA6RjTXOGyp40Xiayo_FwuCaYFgc_KeVfFQ5GPl_GYG5APb9iKziJgIWEH0fPIWwe2hmeRJatHwkXDdT37nwAotogLBCbOohxGtcfbpbM5yWlsFeEOpELQpfrldY8FYhNyYeGOf_Wsp-L-ibd5xmPyYMOWoflIVFISxAd67F6y-H4_SX8vvJbTSrRay3jHDxmidd0kZ75NFNtPDpRK5I65Uzw » empêchée en raison d’un protocole inconnu.`

**Reply by RS-eco** [reply]
Cool, thanks a lot. I managed to register now! ;)
